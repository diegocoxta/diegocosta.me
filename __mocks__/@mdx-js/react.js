module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
