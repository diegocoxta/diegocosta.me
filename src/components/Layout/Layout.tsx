import React from 'react';
import { createGlobalStyle, ThemeContext } from 'styled-components';

import Header from '~/components/Header';
import LanguageSwitcher from '~/components/LanguageSwitcher';
import Footer from '~/components/Footer';

import { useTheme } from '~/hooks/useTheme';
import { dark, light } from '~/themes';

type CustomTheme = typeof dark;

const GlobalStyle = createGlobalStyle<{ theme: CustomTheme }>`
  body {
    background: ${({ theme }) => theme.backgroundColor};
    font-family: 'Source Sans Pro', sans-serif;
    color: ${({ theme }) => theme.textColor};
    margin: 0;
    padding: 0;
  }
`;

export interface LayoutProps {
  children: React.ReactNode;
}

export default function Layout(props: LayoutProps): React.ReactElement {
  const [theme, themeToggler, setMode] = useTheme();

  const themeMode = theme === 'light' ? light : dark;

  return (
    <ThemeContext.Provider value={{ ...themeMode, theme, themeToggler, setMode }}>
      <GlobalStyle />
      <LanguageSwitcher />
      <Header />
      <main>{props.children}</main>
      <Footer />
    </ThemeContext.Provider>
  );
}
