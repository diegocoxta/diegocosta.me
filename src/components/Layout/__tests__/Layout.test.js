import React from 'react';
import { render } from '@testing-library/react';

import Layout from '../Layout';

jest.mock('../../Header', () => () => <p>Header</p>);
jest.mock('../../Footer', () => () => <p>Footer</p>);

describe('<Layout />', () => {
  it('renders properly', () => {
    const { baseElement, queryByText } = render(<Layout>John Doe</Layout>);
    expect(queryByText('John Doe')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
