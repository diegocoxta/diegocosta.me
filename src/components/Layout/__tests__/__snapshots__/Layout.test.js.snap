// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders properly 1`] = `
.c0 {
  max-width: 960px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 40px 0 0 0;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 {
  margin: 0 5px;
  color: #2F3D4F;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 22px;
}

<body>
  <div>
    <section
      class="c0"
    >
      <main>
        <div
          class="c1"
        >
          <a
            aria-label="languageswitcher.toggle languages.pt"
            class="c2"
            href="/about-us"
            language="pt"
          >
            /
            pt
          </a>
          <a
            aria-label="languageswitcher.toggle languages.en"
            class="c2"
            href="/about-us"
            language="en"
          >
            en
          </a>
          <a
            aria-label="languageswitcher.toggle languages.es"
            class="c2"
            href="/about-us"
            language="es"
          >
            es
          </a>
        </div>
      </main>
    </section>
    <p>
      Header
    </p>
    <main>
      John Doe
    </main>
    <p>
      Footer
    </p>
  </div>
</body>
`;
